### PR TITLE
[MODUSERS-412] - Add fields to filtering in user tenants endpoint

### PR DESCRIPTION
**Pull request by Dmitriy-Butramyou**
## Purpose
[MODUSERS-412](https://issues.folio.org/browse/MODUSERS-412) - Add missed fields to filtering in user tenants endpoint

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/UserTenantsAPI.java [hunk 1]
```diff
@@ -83,7 +85,9 @@ private void addWhereClauseArgumentsToCriterion(ArgumentsHolder argumentsHolder,
       TENANT_ID_FIELD, StringUtils.defaultString(argumentsHolder.tenantId()),
       EMAIL, StringUtils.defaultString(argumentsHolder.email()),
       PHONE_NUMBER, StringUtils.defaultString(argumentsHolder.phoneNumber()),
-      MOBILE_PHONE_NUMBER, StringUtils.defaultString(argumentsHolder.mobilePhoneNumber())
+      MOBILE_PHONE_NUMBER, StringUtils.defaultString(argumentsHolder.mobilePhoneNumber()),
+      BARCODE, StringUtils.defaultString(argumentsHolder.barcode()),
+      EXTERNAL_SYSTEM_ID, StringUtils.defaultString(argumentsHolder.externalSystemId())
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/rest/impl/UserTenantsAPI.java:90]: we should add unit tests

Comment (SerhiiNosko) [reply src/main/java/org/folio/rest/impl/UserTenantsAPI.java:90]: looks good, thanks

**Review by SerhiiNosko — APPROVED** [review]

**Review by Abdulkhakimov — APPROVED** [review]

File: src/main/java/org/folio/rest/impl/UserTenantsAPI.java [hunk 2]
```diff
@@ -92,5 +96,5 @@ private void addWhereClauseArgumentsToCriterion(ArgumentsHolder argumentsHolder,
       .forEach(param -> criterion.addCriterion(param, queryOp));
   }
 
-  record ArgumentsHolder(String userId, String username, String tenantId, String email, String phoneNumber, String mobilePhoneNumber) {}
+  record ArgumentsHolder(String userId, String username, String tenantId, String email, String phoneNumber, String mobilePhoneNumber, String barcode, String externalSystemId) {}
```
Comment (imerabishvili) [inline src/main/java/org/folio/rest/impl/UserTenantsAPI.java:99]: Do we have any code style rules in this module? In this case long line creates horizontal scroll.
```
  record ArgumentsHolder(
    String userId,
    String username,
    String tenantId,
    String email,
    String phoneNumber,
    String mobilePhoneNumber,
    String barcode,
    String externalSystemId
  ) {
  }
```

File: src/main/java/org/folio/rest/impl/UserTenantsAPI.java [hunk 3]
```diff
@@ -83,7 +85,9 @@ private void addWhereClauseArgumentsToCriterion(ArgumentsHolder argumentsHolder,
       TENANT_ID_FIELD, StringUtils.defaultString(argumentsHolder.tenantId()),
       EMAIL, StringUtils.defaultString(argumentsHolder.email()),
       PHONE_NUMBER, StringUtils.defaultString(argumentsHolder.phoneNumber()),
-      MOBILE_PHONE_NUMBER, StringUtils.defaultString(argumentsHolder.mobilePhoneNumber())
+      MOBILE_PHONE_NUMBER, StringUtils.defaultString(argumentsHolder.mobilePhoneNumber()),
+      BARCODE, StringUtils.defaultString(argumentsHolder.barcode()),
+      EXTERNAL_SYSTEM_ID, StringUtils.defaultString(argumentsHolder.externalSystemId())
       );
```
Comment (imerabishvili) [inline src/main/java/org/folio/rest/impl/UserTenantsAPI.java:91]: Line has an extra indent.

Comment (Dmitriy-Butramyou) [reply src/main/java/org/folio/rest/impl/UserTenantsAPI.java:91]: added extra indent.

File: src/main/java/org/folio/rest/impl/UserTenantsAPI.java (hunk 2 above)
Comment (Dmitriy-Butramyou) [reply src/main/java/org/folio/rest/impl/UserTenantsAPI.java]: Updated

**Pull request closed without merging**